plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -62,11 +62,10 @@ export class ArgoService implements ArgoServiceApi {
             token,
           );
         } catch (error: any) {
-          this.logger.error(
-            `failed to fetch app data from ${argoInstance.name}: ${String(
-              error,
-            )}`,
-          );
+          return null;
+        }
+
+        if (options.selector && !getArgoAppDataResp.items) {
```

Comment:
I wonder does it make senses to add a test for this?

Reply:
There should be tests around this, yes. I will have a look at it and update.